dvc/config.py

```diff
@@ -24,13 +24,29 @@ def __init__(self, msg):
 
 
 class NoRemoteError(ConfigError):
-    def __init__(self, command):
-        msg = (
-            "no remote specified. Setup default remote with\n"
-            "    dvc remote default <name>\n"
-            "or use:\n"
-            "    dvc {} -r <name>\n".format(command)
-        )
+    def __init__(self, command, *, config=None):
```

Comment:
what is '*' for?

Reply:
It's to make sure `config` can only be used as a keyword argument.

Reply:
The reasoning for it is to prevent the exception from being initialized with some context that might not be a Config.

Reply:
So it is to make sure that we will never use it as a keyword argument? Isn't it kind of protecting the code from the developer?

Reply:
Exactly.

Reply:
Sorry, yeah, I meant that its to make sure that we always use it as a keyword argument.

Reply:
@fabiosantoscode but I still do not understand why would we need that here, what makes `config` so special that it needs to be always used as keyword argument?

Reply:
Nothing makes it special really. I just thought somebody might make a mistake and pass something else as an argument here. But I'll change it back.

Reply:
@fabiosantoscode this is a good point, and we actually have seen few times when wrong arguments are passed due to ordering, but I guess that there is no point in introducing that in this single place if we do differently everywhere else. We just need to be careful when using `kwargs` as `args`.

Reply:
I am +1 on using keyword only args in situations like this.

Reply:
@Suor  So maybe we should enforce this rule throughout the code?

Reply:
@pared @Suor I think we have a ticket for doing that in api.py already. Just need to be consistent, we weren't doing that earlier because py2 didn't support it, IIRC.